app/src/main/java/de/tum/in/tumcampusapp/activities/RoomFinderActivity.java

```diff
@@ -3,39 +3,42 @@
 import android.app.SearchManager;
 import android.content.Intent;
 import android.database.Cursor;
+import android.os.AsyncTask;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.AdapterView;
 import android.widget.AdapterView.OnItemClickListener;
 
+import com.google.common.base.Optional;
+
+import java.io.IOException;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
 
 import de.tum.in.tumcampusapp.R;
-import de.tum.in.tumcampusapp.activities.generic.ActivityForSearching;
+import de.tum.in.tumcampusapp.activities.generic.ActivityForSearchingInBackground;
 import de.tum.in.tumcampusapp.adapters.NoResultsAdapter;
 import de.tum.in.tumcampusapp.adapters.RoomFinderListAdapter;
+import de.tum.in.tumcampusapp.api.TUMCabeClient;
+import de.tum.in.tumcampusapp.auxiliary.NetUtils;
 import de.tum.in.tumcampusapp.auxiliary.RoomFinderSuggestionProvider;
+import de.tum.in.tumcampusapp.auxiliary.Utils;
 import de.tum.in.tumcampusapp.managers.RecentsManager;
-import de.tum.in.tumcampusapp.tumonline.TUMRoomFinderRequest;
-import de.tum.in.tumcampusapp.tumonline.TUMRoomFinderRequestFetchListener;
+import de.tum.in.tumcampusapp.models.tumcabe.RoomFinderRoom;
 import se.emilsjolander.stickylistheaders.StickyListHeadersListView;
 
 /**
  * Activity to show a convenience interface for using the MyTUM room finder.
  */
-public class RoomFinderActivity extends ActivityForSearching implements TUMRoomFinderRequestFetchListener, OnItemClickListener {
-
-    // HTTP client for sending requests to MyTUM roomFinder
-    private TUMRoomFinderRequest roomFinderRequest;
+public class RoomFinderActivity extends ActivityForSearchingInBackground<List<RoomFinderRoom>>
+        implements OnItemClickListener {
 
     private RecentsManager recentsManager;
     private StickyListHeadersListView list;
     private RoomFinderListAdapter adapter;
 
+    private AsyncTask<String, Void, List<RoomFinderRoom>> searchRoomAsyncTask;
```

Comment:
No it is not used. I declare this variable before changing the subclass from ActivityForSearching to ActivityForSearchingInBackground. 
It is should be deleted.